gax-httpjson/src/main/java/com/google/api/gax/httpjson/ManagedHttpJsonChannel.java

```
@@ -43,62 +46,139 @@
+                }
+
+                // Populate HTTP path and query parameters.
+                Map<String, String> pathParams =
+                    requestBuilder.getPathParams(request, methodDescriptor.pathParams());
+                PathTemplate pathPattern =
+                    PathTemplate.create(methodDescriptor.endpointPathTemplate());
+                String relativePath = pathPattern.instantiate(pathParams);
+                GenericUrl url = new GenericUrl(endpoint + relativePath);
+                Map<String, List<String>> queryParams =
+                    requestBuilder.getQueryParams(request, methodDescriptor.queryParams());
+                for (String queryParam : methodDescriptor.queryParams()) {
+                  if (queryParams.containsKey(queryParam) && queryParams.get(queryParam) != null) {
+                    url.set(queryParam, queryParams.get(queryParam));
+                  }
+                }
+
+                HttpRequest httpRequest =
+                    requestFactory.buildRequest(
+                        methodDescriptor.httpMethod().name(), url, jsonHttpContent);
+
+                for (HttpJsonHeaderEnhancer enhancer : headerEnhancers) {
+                  enhancer.enhance(httpRequest.getHeaders());
+                }
+                httpRequest.setParser(new JsonObjectParser(jsonFactory));
 
-              GenericData responseData = response.parseAs(GenericData.class);
-              // TODO convert GenericData to response type
+                HttpResponse httpResponse = httpRequest.execute();
 
-              responseFuture.set(responseData);
+                ResponseT response =
+                    methodDescriptor.parseResponse(
+                        new InputStreamReader(httpResponse.getContent()));
+                responseFuture.set(response);
 
+              } catch (IOException e) {
+                e.printStackTrace(System.err);
```

Don't hide this exception

---

Moved it to the responseFuture.